test/e2e/operator_test.go

```diff
@@ -1199,6 +1203,183 @@ func TestAWSLoadBalancerControllerOpenAPIValidation(t *testing.T) {
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("failed to ensure echo ingress %s: %v", echoIng.Name, err)
+	}
+	defer func() {
+		waitForDeletion(context.TODO(), t, kubeClient, echoIng, defaultTimeout)
+	}()
+
+	_, err = getIngress(context.TODO(), t, kubeClient, defaultTimeout, ingName)
+	if err != nil {
+		t.Fatalf("did not get load balancer hostname for ingress: %v", err)
+	}
+
+	// Save a copy of the original infra Config, to revert changes before exiting.
+	originalInfra := infra.DeepCopy()
+	defer func() {
+		err := updateInfrastructureConfigStatusWithRetryOnConflict(t, 5*time.Minute, kubeClient, func(infra *configv1.Infrastructure) *configv1.Infrastructure {
+			infra.Status = originalInfra.Status
+			return infra
+		})
+		if err != nil {
+			t.Logf("Unable to remove changes from the infraConfig, possible corruption of test environment: %v", err)
+		}
+	}()
+
+	// Update infrastructure status with initialInfraTags
+	initialInfraTags := []configv1.AWSResourceTag{
+		{Key: "plat-key1", Value: "plat-value1"},
+		{Key: "conflict-key1", Value: "plat-value2"},
+		{Key: "conflict-key2", Value: "plat-value3"},
+	}
+	t.Logf("Updating AWS ResourceTags in the cluster infrastructure config: %v", initialInfraTags)
+	err = updateInfrastructureConfigStatusWithRetryOnConflict(t, 5*time.Minute, kubeClient, func(infra *configv1.Infrastructure) *configv1.Infrastructure {
+		if infra.Status.PlatformStatus == nil {
+			infra.Status.PlatformStatus = &configv1.PlatformStatus{}
+		}
+		if infra.Status.PlatformStatus.AWS == nil {
+			infra.Status.PlatformStatus.AWS = &configv1.AWSPlatformStatus{}
+		}
+		infra.Status.PlatformStatus.AWS.ResourceTags = initialInfraTags
+		return infra
+	})
+	if err != nil {
+		t.Errorf("failed to update infrastructure status: %v", err)
+	}
+
+	// wait for the deployment to restart and become available
+	if err := waitForDeploymentRollout(context.TODO(), t, kubeClient, defaultTimeout, deploymentName, depGeneration); err != nil {
+		t.Fatalf("deployment did not roll out within timeout: %v", err)
+	}
+	if err := waitForDeploymentStatusCondition(context.TODO(), t, kubeClient, defaultTimeout, deploymentName, expected...); err != nil {
+		t.Fatalf("did not get expected available condition for deployment: %v", err)
+	}
+
+	// get the updated deployment
+	if err := kubeClient.Get(context.TODO(), deploymentName, dep); err != nil {
+		t.Fatalf("failed to get deployment %s: %v", deploymentName.Name, err)
+	}
+	depGeneration = dep.Generation
+
+	t.Logf("Testing aws tags present in alb instance and infra status (initialInfraTags)")
+	expectedTags := map[string]string{
+		"conflict-key1": "op-value2", "conflict-key2": "op-value3", "op-key1": "op-value1", "plat-key1": "plat-value1",
+	}
+	// Check `--default-tags` container argument
+	assertContainerArgFromDeployment(t, dep, awsLoadBalancerControllerContainerName, "--default-tags", convertTagsMapToString(expectedTags))
```

Comment:
TBH, I think from the ALBO's standpoint confirming `--default-tags` is also essential to check on a real cluster, because this is what we're modifying through our code, even though we can expect the same results directly from AWS. 

For me:

- `--default-tags` arg confirms ALBO's design 
- Tags on the load balancer instance confirm's ALBC's design 

Moreover, this function only checks an expected container argument value on a deployment, which should be a cheap execution without any retries. Not sure how much CI time and CPU cycle we can save, by removing this check. If this can make a significant difference, then we can remove it later may be.